exposed-tests/src/test/kotlin/org/jetbrains/exposed/sql/tests/shared/dml/UpsertTests.kt

```diff
@@ -325,6 +349,60 @@ class UpsertTests : DatabaseTestsBase() {
         }
     }
 
+    @Test
+    fun testUpsertWithManualUpdateUsingInsertValues() {
+        val tester = object : Table("tester") {
+            val id = integer("id").uniqueIndex()
+            val word = varchar("name", 64)
+            val count = integer("count").default(1)
+        }
+
+        withTables(excludeSettings = TestDB.ALL_H2_V1, tester) {
+            tester.insert {
+                it[id] = 1
+                it[word] = "Word A"
+            }
+            assertEquals(1, tester.selectAll().single()[tester.count])
+
+            // H2_Mysql & H2_Mariadb syntax does not allow VALUES() syntax to come first in complex expression
+            // Syntax must be column=(1 + VALUES(column)), not column=(VALUES(column) + 1)
+            tester.upsert {
+                it[id] = 1
+                it[word] = "Word B"
+                it[count] = 9
+
+                it.onUpdate { update ->
+                    update[count] = intLiteral(100) times insertValue(count)
+                }
```

Comment:
This test highlights the duality of this form of `onUpdate`.
The inner function will act using the same pattern as an `InsertStatement`, for example.

So with `upsert()`, the statement is passed as the argument to the lambda block and the new function (like the setters above it) will only be accessible by `it.onUpdate()`. This leads to the implicit parameter being shadowed, which is why I've named it `update` here.

With `batchUpsert()` below, the statement is the receiver, so it's accessed as `this.onUpdate()` and the inner parameter doesn't need to be named to be less confusing.